api/api_controller.go

```diff
@@ -51,6 +53,8 @@ const (
 	UserContextKey        contextKey = "user"
 )
 
+var cfg = config.NewConfig()
```

Comment:
While this would work, creating a new config object here creates a coupling between the api controller and the config, which is not ideal. Since you don't need the entire config, just the blockstore type for now, I would just take it explictly from `deps.BlockAdapter.BlockstoreType()`
(you get the `deps` once you add the `setupRequest` call like @arielshaqed suggested.

Reply:
I changed it. Now it uses deps, however it works even without the setupRequest so I'm a little confused